File: src/uz/efir/muazzin/MuazzinAdapter.java
package uz.efir.muazzin;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

public class MuazzinAdapter extends FragmentStatePagerAdapter {
    private final Context mContext;

    public MuazzinAdapter(Context context, FragmentManager fm) {
        super(fm);
        mContext = context;
    }

    @Override
    public Fragment getItem(int position) {
        switch (position) {
        case 0:
            return new PrayerTimesFragment();
        case 1:
            return new QiblaCompassFragment();
        default:
            return null;
        }
    }

    @Override
    public int getCount() {
        // time table + compass
        return 2;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        switch (position) {
        case 0:
            return mContext.getString(R.string.today);
        case 1:
            return mContext.getString(R.string.qibla);
        default:
            return null;
        }
    }
}

File: app/src/main/kotlin/uz/efir/muazzin/MuazzinAdapter.kt
package uz.efir.muazzin

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter

class MuazzinAdapter(activity: FragmentActivity) : FragmentStateAdapter(activity) {
    override fun getItemCount(): Int = 2

    override fun createFragment(position: Int): Fragment = when (position) {
        1 -> QiblaCompassFragment()
        else -> PrayerTimesFragment()
    }
}
